bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -554,6 +560,22 @@ func (r *KubeadmConfigReconciler) MachineToBootstrapMapFunc(o handler.MapObject)
 	return result
 }
 
+// MachinePoolToBootstrapMapFunc is a handler.ToRequestsFunc to be used to enqueue
+// request for reconciliation of KubeadmConfig.
+func (r *KubeadmConfigReconciler) MachinePoolToBootstrapMapFunc(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	m, ok := o.Object.(*clusterv1.MachinePool)
+	if !ok {
+		return nil
+	}
+	if m.Spec.Template.Spec.Bootstrap.ConfigRef != nil && m.Spec.Template.Spec.Bootstrap.ConfigRef.GroupVersionKind().GroupKind() == bootstrapv1.GroupVersion.WithKind("KubeadmConfig").GroupKind() {
```

Comment:
Nit: assign `m.Spec.Template.Spec.Bootstrap.ConfigRef` to a variable so it's a little more readable